cli/eva_sub_cli.py

```diff
@@ -1,20 +1,70 @@
+                           container, docker, sub_config)
+
+
 if __name__ == "__main__":
-    argparse = ArgumentParser(description='EVA Submission CLI')
-    argparse.add_argument('--submission-dir', required=True, type=str,
-                          help='Full path to the submission directory where all submission info is/will be stored')
-    argparse.add_argument('--resume', action='store_true', default=False, help='resume an existing submission')
-    args = argparse.parse_args()
-
-    logging_config.add_stdout_handler()
-
-    submitter = StudySubmitter()
-    if args.resume:
-        submitter.upload_submission(args.submission_dir)
-    else:
-        submitter.submit(args.submission_dir)
+    argparser = ArgumentParser(description='EVA Submission CLI - validate and submit data to EVA')
+    argparser.add_argument('--task', required=True, choices=[VALIDATE, SUBMIT, RESUME],
+                           help='Select a task to perform')
+    argparser.add_argument('--submission_dir', required=True, type=str,
+                           help='Full path to the directory where all processing will be done and submission info is/will be stored')
+    argparser.add_argument("--docker_path", required=False, help="Full path to the docker installation, "
+                                                "not required if docker is available in the PATH environment variable")
+    argparser.add_argument("--container_name", help="Name of the docker container", required=False)
+    argparser.add_argument("--vcf_files_mapping", required=False,
+                           help="csv file with the mappings for vcf files, fasta and assembly report")
+    group = argparser.add_mutually_exclusive_group(required=False)
+    group.add_argument("--metadata_json",
+                       help="Json file that describe the project, analysis, samples and files")
+    group.add_argument("--metadata_xlsx",
+                       help="Excel spreadsheet  that describe the project, analysis, samples and files")
+
+    args = argparser.parse_args()
+
+    # load config
+    config_file_path = os.path.join(args.submission_dir, SUB_CLI_CONFIG_FILE)
+    sub_config = WritableConfig(config_file_path, version=__version__)
+
+    if args.task == RESUME:
+        submitter = StudySubmitter(args.submission_dir, submission_config=sub_config)
+        submitter.upload_submission()
+
+    if args.task == VALIDATE or args.task == SUBMIT:
+        if not args.vcf_files_mapping:
+            raise Exception(f"Please provide a CSV file which provides VCF and corresponding assembly information "
+                            f"(use the --vcf_files_mapping switch)")
+        if not args.metadata_json and not args.metadata_xlsx:
+            raise Exception(f"Please provide the file that describes the project, analysis, samples and files "
+                            f"using either --metadata_json or --metadata_xlsx")
```

Comment:
This could probably be improved in a subsequent ticket but the upload does need to know where the VCF and metadata is.

Reply:
updated